doc/book/cookbook/setting-module-specific-layout.md

```diff
@@ -0,0 +1,85 @@
+# Setting module-specific Layout
+
+The following example shows how to set a template for the layout based on a
+module name in a zend-mvc based application. The example uses a listener that
+listen on the
+[`Zend\Mvc\MvcEvent::EVENT_RENDER` event](https://docs.zendframework.com/zend-mvc/mvc-event/#mvceventevent_render-render)
+and uses the
+[`Zend\Router\RouteMatch` object](https://docs.zendframework.com/zend-mvc/routing/#routing)
+to get the called module from current request.
+
+## Create Listener
+
+Create a listener as separate class, e.g.
+`module/Admin/src/Listener/LayoutListener.php`:
+
+```php
+namespace Admin\Listener;
+
+use Zend\EventManager\AbstractListenerAggregate;
+use Zend\EventManager\EventManagerInterface;
+use Zend\Mvc\MvcEvent;
+
+class LayoutListener extends AbstractListenerAggregate
+{
+    public function attach(EventManagerInterface $events, $priority = 1)
+    {
+        $this->listeners[] = $events->attach(
+            MvcEvent::EVENT_RENDER,
+            [
+                $this,
+                'setLayout',
+            ]
```

Comment:
Since the second argument here is supposed to be a callback, I'd argue that it should not be split over multiple lines, but rather be in the form `[$this, 'setLayout']` on one line.